pkg/controllers/redis_acls.go

```diff
@@ -0,0 +1,140 @@
+package controllers
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	redis "github.com/scaleway/scaleway-sdk-go/api/redis/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"k8s.io/api/core/v1"
+	klog "k8s.io/klog/v2"
+)
+
+func (c *NodeController) syncRedisACLs(nodeName string) error {
+	if len(c.redisIDs) == 0 {
+		return nil
+	}
+
+	var node *v1.Node
+
+	retryOnError := false
+
+	nodeObj, exists, err := c.indexer.GetByKey(nodeName)
+	if err != nil {
+		klog.Errorf("could not get node %s by key: %v", nodeName, err)
+		return err
+	}
+	if exists {
+		var ok bool
+		node, ok = nodeObj.(*v1.Node)
+		if !ok {
+			klog.Errorf("could not get node %s from object", nodeName)
+			return fmt.Errorf("could not get node %s from object", nodeName)
+		}
+	}
+
+	dbAPI := redis.NewAPI(c.scwClient)
+
+	for _, redisID := range c.redisIDs {
+		klog.Infof("whitelisting IP on node %s on redis instance %s", nodeName, redisID)
+
+		id, zone, err := getRegionalizedID(redisID)
+		if err != nil {
+			klog.Errorf("could not get id and zone from %s: %v", redisID, err)
+			continue
+		}
+
+		dbInstance, err := dbAPI.GetCluster(&redis.GetClusterRequest{
+			Zone:      scw.Zone(zone),
+			ClusterID: id,
+		})
+		if err != nil {
+			klog.Errorf("could not get redis instance %s: %v", id, err)
+			continue
+		}
+
+		var rule *redis.ACLRule
+
+		for _, acl := range dbInstance.ACLRules {
+			if *acl.Description == nodeName {
+				rule = acl
+				break
+			}
+		}
+
+		if !exists && rule != nil {
+			err := dbAPI.DeleteACLRule(&redis.DeleteACLRuleRequest{
+				Zone:  dbInstance.Zone,
+				ACLID: rule.IP.String(),
+			})
+			if err != nil {
+				klog.Errorf("could not delete acl rule for node %s on redis instance %s: %v", nodeName, dbInstance.ID, err)
+				retryOnError = true
+			}
+			continue
+		}
+
+		var nodePublicIP net.IP
+
+		if os.Getenv(NodesIPSource) == NodesIPSourceKubernetes {
+			for _, addr := range node.Status.Addresses {
+				if addr.Type == v1.NodeExternalIP {
+					nodePublicIP = net.ParseIP(addr.Address)
+					if len(nodePublicIP) == net.IPv6len {
+						// prefer ipv4 over ipv6 since Database are only accessible via ipv4
```

Comment:
This won't work if the ipv4 is before the ipv6 in the array, maybe just break if it's an ipv4 instead

Reply:
good catch 👍  
fixed on redis & rdb ACLs